ios/FluentUI/Avatar/AvatarTokens.swift

```diff
@@ -6,179 +6,274 @@
+            return aliasTokens.backgroundColors[.neutralDisabled]
+        case .outlinedPrimary:
+            return .init(light: globalTokens.brandColors[.tint10].light, dark: globalTokens.neutralColors[.grey78])
         }
     }
 
-    var size: MSFAvatarSize {
-        didSet {
-            if oldValue != size {
-                updateForCurrentTheme()
-            }
+    /// The color of the gap between the ring and `Avatar`.
+    open var ringGapColor: DynamicColor { aliasTokens.backgroundColors[.neutral1] }
+
+    /// The thickness of the ring around the `Avatar`.
+    open var ringThickness: CGFloat {
+        switch size {
+        case .xsmall, .small:
+            return globalTokens.borderSize[.thin]
+        case .medium, .large, .xlarge:
+            return globalTokens.borderSize[.thick]
+        case .xxlarge:
+            return globalTokens.borderSize[.thicker]
+        }
+    }
+
+    /// The gap between the `Avatar` and its ring.
+    open var ringInnerGap: CGFloat {
+        switch size {
+        case .xsmall, .small, .medium, .large, .xlarge:
+            return globalTokens.borderSize[.thick]
+        case .xxlarge:
+            return globalTokens.borderSize[.thicker]
+        }
+    }
+
+    /// The gap around the ring around the `Avatar`.
+    open var ringOuterGap: CGFloat {
+        switch size {
+        case .xsmall, .small, .medium, .large, .xlarge:
+            return globalTokens.borderSize[.thick]
+        case .xxlarge:
+            return globalTokens.borderSize[.thicker]
         }
     }
 
-    init(style: MSFAvatarStyle,
-         size: MSFAvatarSize) {
-        self.style = style
-        self.size = size
+    /// The size of the presence icon.
+    open var presenceIconSize: CGFloat {
+        switch size {
+        case .xsmall:
+            return 0
+        case .small, .medium:
+            return globalTokens.iconSize[.xxxSmall]
+        case .large, .xlarge:
+            return globalTokens.iconSize[.xxSmall]
+        case .xxlarge:
+            return globalTokens.iconSize[.small]
+        }
+    }
 
-        super.init()
+    /// The thickness of the outline around the presence icon.
+    open var presenceIconOutlineThickness: CGFloat {
+        switch size {
+        case .xsmall:
+            return globalTokens.borderSize[.none]
+        case .small, .medium, .large, .xlarge, .xxlarge:
+            return globalTokens.borderSize[.thick]
+        }
+    }
 
-        self.themeAware = true
+    /// The color of the outline around the presence.
+    open var presenceOutlineColor: DynamicColor { aliasTokens.backgroundColors[.neutral1] }
 
-        updateForCurrentTheme()
+    /// The color for the background calculated using a hash of the strings.
+    open var backgroundCalculatedColorOptions: [DynamicColor] {
+        // will sync and update once these are checked in
+        return [.init(light: globalTokens.brandColors[.tint40].light, dark: globalTokens.brandColors[.shade30].dark),
```

Comment:
Added in a bunch of temp colors, but at least they're the right tint/shade, so the actual color change should be minimal.